server/tinylicious/src/runner.ts

```diff
@@ -34,21 +35,21 @@ export class TinyliciousRunner implements utils.IRunner {
         private readonly storage: IDocumentStorage,
         private readonly mongoManager: MongoManager,
         private readonly contentCollection: ICollection<any>,
-    ) {
-    }
+    ) {}
 
-    public start(): Promise<void> {
+    public async start(): Promise<void> {
         this.runningDeferred = new Deferred<void>();
 
-        // Create the HTTP server and attach alfred to it
-        const alfred = app.create(
-            this.config,
-            this.storage,
-            this.mongoManager);
+        try {
+            await this.ensurePortIsFree();
+        } catch (error) {
+            this.runningDeferred.reject(error);
```

Comment:
this won't end control flow? should we just let ensurePort throw?

Reply:
Agreed. Will make it so. In the future we can probably ask whether the individual would like to continue and ask for input (and suggest the free port returned by ensurePortIsFree).